### PR TITLE
[typescript] fix: `explode: true` should yield appended `query` params

### PR DESCRIPTION
**Pull request by joscha**
Currently, when a query param is defined as:

```yaml
        - name: context
          in: query
          required: true
          explode: true
          schema:
            type: array
            items:
              type: string
```
for a `context` of `['a','b','c']` it serializes the query incorrectly as:
`?context=a,b,c`
however, due to `explode: true` it should be:
`?context=a&context=b&context=c`.

The change in this PR makes sure that any query params with `explode: true` are serialized correctly.

It also fixes the exploded serialization for `object` types.

See example table here: https://swagger.io/docs/specification/serialization/#Query%20Parameters

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee): @TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript/api/api.mustache [hunk 1]
```diff
@@ -66,7 +66,27 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
 
         // Query Params
         if ({{paramName}} !== undefined) {
+            {{#isPrimitiveType}}
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api/api.mustache:69]: we're leaving primitives alone, as their `explode: true` and `explode: false` is the same.

File: modules/openapi-generator/src/main/resources/typescript/api/api.mustache [hunk 2]
```diff
@@ -66,7 +66,27 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
...
+                requestContext.appendQueryParam("{{baseName}}", serializedParam);
+            }
+            {{/isArray}}
+            {{^isArray}}
+            for (const key of Object.keys(serializedParams)) {
+                requestContext.appendQueryParam(key, serializedParams[key]);
+            }
+            {{/isArray}}
+            {{/isExplode}}
+            {{^isExplode}}
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api/api.mustache:86]: for unexploded non-primitives we keep the current serialization `name=value1,value2,value3`

File: modules/openapi-generator/src/main/resources/typescript/api/api.mustache [hunk 3]
```diff
@@ -66,7 +66,27 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
...
+            {{/isPrimitiveType}}
+            {{^isPrimitiveType}}
+            {{#isExplode}}
+            const serializedParams = ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}");
+            {{#isArray}}
+            for (const serializedParam of serializedParams) {
+                requestContext.appendQueryParam("{{baseName}}", serializedParam);
+            }
+            {{/isArray}}
+            {{^isArray}}
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api/api.mustache:80]: this is the new exploded serialization for objects

File: modules/openapi-generator/src/main/resources/typescript/api/api.mustache [hunk 4]
```diff
@@ -66,7 +66,27 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
...
         // Query Params
         if ({{paramName}} !== undefined) {
+            {{#isPrimitiveType}}
             requestContext.setQueryParam("{{baseName}}", ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}"));
+            {{/isPrimitiveType}}
+            {{^isPrimitiveType}}
+            {{#isExplode}}
+            const serializedParams = ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}");
+            {{#isArray}}
+            for (const serializedParam of serializedParams) {
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript/api/api.mustache:76]: this is the new exploded serialization for arrays

**Review by macjohnny — APPROVED** [review]
thanks for your contribution!

**Comment by bodograumann** [comment]
For anyone looking here, I just want to mention that this broke enum serialization in query parameter, but it is now fixed on master: https://github.com/OpenAPITools/openapi-generator/pull/20210

**Pull request merged**